### ISSUE TITLE
Inappropriate fond size in reference diagram C.1.0

### ISSUE THREAD

**Issue by bzinchenko** [bpmn-miwg/bpmn-miwg-test-suite#697]
Diagram C.1.0 has font set inappropriately large:

```
<bpmndi:BPMNLabelStyle id="Bpmndi_BPMNLabelStyle_M44qwJ1_EeS1-pEyeWEPig">
  <omgdc:Font name="Segoe UI" size="16.35"/>
</bpmndi:BPMNLabelStyle>

```
One can clearly see that font is two times bigger than in all other examples. Compare for instance with C.1.1: 

```
<bpmndi:BPMNLabelStyle id="Bpmndi_BPMNLabelStyle_cd6FsLH1EeSuDf0W70XLGw">
   <omgdc:Font name="Segoe UI" size="9.0"/>
</bpmndi:BPMNLabelStyle>

```

If displayed proportionally, this font creates distortions on diagram as illustrated in issue #686. Many tools simply ignore font sizes, which caused this issue not noticed earlier.

**Comment by benmichel** [comment]
Boris, I beg to differ. According to the outcome of the discussions around font size in #497 we decided that pixel should be the font size unit. C.1.0 is actually the only referece diagram to respect that and have a correct font size.

**Comment by bzinchenko** [comment]
Michel, thank you for the reply. If C.1.0 diagram is correct, then **all** other diagrams are incorrect and should be normalized for font sizes. It is not acceptable when diagrams have font sizes, which differ almost twice and still appear same visually! It does not matter, in which units we measure fonts. It is just issue of observing a relative proportion.

**Comment by bzinchenko** [comment]
To be sure, I double checked the code that it renders fonts exactly in pixels:

`Font font = new Font(ft.name.Value, (float)ft.size.Value, fontStyle, GraphicsUnit.Pixel);`

Here GraphicsUnit.Pixel exactly corresponds to pixel resolution in font units on rendering.
https://msdn.microsoft.com/en-us/library/c8dy0032(v=vs.110).aspx
https://msdn.microsoft.com/en-us/library/system.drawing.font.unit(v=vs.110).aspx
https://msdn.microsoft.com/en-us/library/system.drawing.graphicsunit(v=vs.110).aspx

To check it additionally, I drew red rectangles for label boundaries as in our test suite and compared with label dimensions written in XML. Label sizes in XML are 17 pixels for some elements:

` <omgdc:Bounds height="17.0" width="27.0" x="548.0" y="540.0"/>`

On attached pictures one can see that rendered font is roughly same 17 pixels overall height as respective labels, which proves that it was rendered correctly.

![c 1 0](https://cloud.githubusercontent.com/assets/5006818/20306525/6587f262-ab43-11e6-8342-511da44b5b22.png)

**Comment by benmichel** [comment]
As you say, I agree

> all other diagrams are incorrect and should be normalized for font sizes

and that was the conclusion of issue #497 

If all reference files appear the same size despite different font units, it is because all reference files were not realized by the same tools. C.1.0 was originaly done by W4 BPMN+ composer which use Pixel as font size. The other ones were were done originally by tools using Point as font unit.

I am not sur to understand your last comment but I can say that C.1.0 reference model use Pixel as font unit, the size of the label boxes is serealized accordingly and the bounds overlay is correct (except for lane and pool, but this is known a bug on W4 BPMN+ composer)

![bounds](https://cloud.githubusercontent.com/assets/8003851/20311804/67475774-ab51-11e6-908b-672a6aefda04.PNG)

**Comment by bzinchenko** [comment]
Michel, I believe that font size on your diagram is incorrect. It can be easily verified by just measuring it in pixels in any image editor. Enough to open file and read cursor position difference between top and bottom coordinate of any text line in a label. I suppose real font size in pixels is somewhat close to standard 9 pixels found on other diagrams. I also believe that most other tools simply ignore font sizes in rendering.

**Comment by benmichel** [comment]
Hi Boris,

I maintain that font size is correct since you have to take in account special characters such as Î or Ç. Having a font size of x number of pixels doesn't mean each character in this font have a size of x number of pixels, it means that the size from the lowest point of characters like Ç to the highest point of characters like Î is x number of pixels. 

I verified it with a measuring tool for the same font as in the diagram, and it is indeed 17 pixels.

![special](https://cloud.githubusercontent.com/assets/8003851/21179425/52a47e5a-c1f4-11e6-85f3-1a4742888b1f.PNG)

![capture d ecran 2016-12-14 a 11 57 28](https://cloud.githubusercontent.com/assets/8003851/21179489/9f5eecf8-c1f4-11e6-92be-18f75fcb1b5b.png)

![capture d ecran 2016-12-14 a 11 57 18](https://cloud.githubusercontent.com/assets/8003851/21179480/973db9aa-c1f4-11e6-93e9-a724bc2529e3.png)

**Comment by bzinchenko** [comment]
Michel, thank you for reply. I suppose that the difference is due to default resolution misfit between standard web canvas and desktop canvas on Windows. Does your tool use explicit resolution written in BPMN file? I will inspect this issue further and provide here my test results.

**Comment by benmichel** [comment]
Boris,

If by "resolution" you mean the attribute from the DI:Diagram element, then no, our tool does not use that attribute at the moment.
Note that the above measurements were done on Google Chrome

**Comment by bzinchenko** [comment]
Michel, yes, I mean exactly DI:Diagram element. I m afraid that without using this setting it will be simply impossible to ensure compatibility between web and desktop platforms on font rendering.

**Comment by benmichel** [comment]
Yes, we talked about it in issue #497 about labels and font. The conclusion was that we should use pixel unit and the resolution attribute : https://github.com/bpmn-miwg/bpmn-miwg-test-suite/issues/497#issuecomment-157812919
